django_kafka/tests/test_registry.py

```diff
@@ -61,19 +97,32 @@ def test_retry_consumer_registered(self, mock_retry_consumer_build):
         self.assertCountEqual(list(registry), [key_a, retry_key_a, key_b])
         self.assertIs(registry[retry_key_a], retry_consumer_mock)
 
-    def test_get_key(self):
-        consumer_cls_a = self._get_consumer_cls("ConsumerA", "group_a")
-        consumer_cls_b = self._get_consumer_cls("ConsumerB", "group_b")
-        registry = ConsumersRegistry()
 
-        key_a = registry.get_key(consumer_cls_a)
-        self.assertEqual(
-            key_a,
-            f"{consumer_cls_a.__module__}.{consumer_cls_a.__name__}",
+class ConnectorsRegistryTestCase(SimpleTestCase):
+    def _gen_cls(self, name) -> Type[Connector]:
+        return type[Connector](
+            name,
+            (Connector,),
+            {"config": {}},
         )
 
-        key_b = registry.get_key(consumer_cls_b)
-        self.assertEqual(
-            key_b,
-            f"{consumer_cls_b.__module__}.{consumer_cls_b.__name__}",
-        )
+    def test_get_key(self):
+        connector_cls = self._gen_cls("ConnectorCls")
+        registry = ConnectorsRegistry()
+
+        key = registry.get_key(connector_cls)
+        self.assertEqual(key, connector_cls().name)
+
+    def test_registering_same_key_not_allowed(self):
+        registry = ConnectorsRegistry()
+        connector_a = self._gen_cls("ConnectorA")
+        # give the same class name to the connector to generate same key
+        connector_b = self._gen_cls("ConnectorA")
```

Comment:
We already know same keys are rejected from the test on the base class, right? 

So really this just needs to test the overridden implementation of `get_key`.

This test also relies on knowing that the `.name` property is generated from the classname, which this registry test shouldn't know implementation details of the connector class. If we change the implementation of `Connector.name`, then the test would fail when it shouldn't. IMO test just needs to be:

```
def test_get_key(self):
   ....

   self.assertEqual(registry.get_key(connector_a), connector_a.name)
   self.assertEqual(registry.get_key(connector_b), connector_b.name)
```

Since we are only concerned with the return value of the name and not the implementation.

Reply:
This makes sure that overwritten classes still preserve the initial behavior. It is fine to know about the class because the registry is made for this class and if class api fails the registry should fail as well.